pkg/ddl/placement_policy_test.go

```diff
@@ -2401,3 +2412,21 @@ func TestRecoverTableWithPlacementPolicy(t *testing.T) {
 		" PARTITION `p2` VALUES LESS THAN (10000))"))
 	checkExistTableBundlesInPD(t, dom, "test", "tp3")
 }
+
+func TestPartitionByWithLabels(t *testing.T) {
+	store, do := testkit.CreateMockStoreAndDomain(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+
+	tk.MustExec("create placement policy pp1 primary_region='r1' regions='r1,r2'")
+
+	tk.MustExec(`CREATE TABLE t1 (id INT)`)
+	tk.MustExec(`ALTER TABLE t1 placement policy pp1`)
+	tk.MustExec(`ALTER TABLE t1 PARTITION BY HASH (id) PARTITIONS 3`)
+	tk.MustQuery("show create table t1").Check(testkit.Rows("" +
+		"t1 CREATE TABLE `t1` (\n" +
+		"  `id` int(11) DEFAULT NULL\n" +
+		") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin /*T![placement] PLACEMENT POLICY=`pp1` */\n" +
+		"PARTITION BY HASH (`id`) PARTITIONS 3"))
+	checkExistTableBundlesInPD(t, do, "test", "t1")
```

Comment:
We can add more test cases:

1. altering a partitioned table, for example:

```
CREATE TABLE tp(id int) placement policy p1
PARTITION BY RANGE(id) (
    PARTITION p1 VALUES LESS THAN (10) placement policy p2,
    PARTITION p2 VALUES LESS THAN MAXVALUE
);

alter table tp
PARTITION BY RANGE(id) (
    PARTITION p1 VALUES LESS THAN (1),
    PARTITION p2 VALUES LESS THAN (2) placement policy p2,
    PARTITION p3 VALUES LESS THAN MAXVALUE
);
```

2. check the old rule still exists before GC
3. check the old rules are deleted after GC
4. the rollback workflow should delete bundles

Reply:
I have added several tests, both for normal flow as well as rollback cases with error injection. Please review and see if there are any more test you would like to see? :)